### PR TITLE
Fix auto detection of default shell

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
In the current state, the default shell detector does not work correctly and in case shell is not specified in `/etc/passwd` file - then user won't get their terminal. You can easily check it on https://che.openshift.io/ by creating any workspace and try to open terminal in che-machine-exec container. You should face an issue when widget is opened but nothing happens.

In this PR, `sh` command is replaced with `sh -c exit 0`, so in case sh is not available - we'll get error. In case sh is available - it will be finished with 0 code.
Also, to avoid accidental hanging - info execs now are timed out in 3 seconds ( experimental value)
Before that change, on minishift terminal was opened 8 of 10 times in my case. With them 20\20.

In addition, it improves error handling and propagating.
Now, we are able to set `LOG_LEVEL=debug` and will see detailed output how things happen

It's tested and the image is available by `sleshchenko/che-machine-exec:sh-fix`

P.S. Thank @AndrienkoAleksandr for helping me a lot in investigation cloud-shell related issues

### What does it fix?
https://github.com/eclipse/che/issues/16012

### CODE REVIEW

**Comment by AndrienkoAleksandr** [comment]
Looks some tests fails

**Comment by sleshchenko** [comment]
Now it's tested with all images in https://github.com/AndrienkoAleksandr/shellImages

**Comment by amisevsk** [comment]
Also, after reviewing this PR I'm a big +1 for whoever wants to migrate to go modules :)

File: exec/kubernetes_exec_manager.go [hunk 1]
```diff
@@ -92,13 +98,22 @@ func (manager *KubernetesExecManager) Create(machineExec *model.MachineExec) (ex
...
+			logrus.Printf("%s is successfully initialized in auto discovered container %s/%s", machineExec.Cmd,
+				containerInfo.PodName, containerInfo.ContainerName)
+			return machineExec.ID, nil
 		}
-	}
 
-	return -1, err
+		var containers []string
+		for _, c := range containersInfo {
+			containers = append(containers, c.PodName+"\\"+c.PodName)
```
[the remark this answers is not in the document]
Comment (sleshchenko) [reply exec/kubernetes_exec_manager.go]: :scream:

File: api/jsonrpc/handlers.go [hunk 2]
```diff
@@ -51,8 +52,15 @@ func jsonRpcCreateExec(_ *jsonrpc.Tunnel, params interface{}, t jsonrpc.RespTran
 	healthWatcher.CleanUpOnExitOrError()
 
 	if err != nil {
-		log.Println("Unable to create machine exec. Cause: ", err.Error())
+		logrus.Errorf("Unable to initialize terminal. Cause: %s", err.Error())
 		t.SendError(jsonrpc.NewArgsError(err))
+		return
+	}
+
+	if id == -1 {
```
[the remark this answers is not in the document]
Comment (sleshchenko) [reply api/jsonrpc/handlers.go]: You're right, but previously there was such a situation by mistake. There was `return -1, err`, where err is a return variable with nil value. So, I put it here just to be safer.

[the remark this answers is not in the document]
Comment (amisevsk) [reply api/jsonrpc/handlers.go]: :+1:

**Pull request merged**